pybossa/api/__init__.py

```diff
@@ -211,8 +211,21 @@ def new_task(project_id, task_id=None):
                 else:
                     # user returning back for the same task
                     # original presented time has not expired yet
-                    # to continue original presented time, extend expiry
-                    guard.extend_task_presented_timestamp_expiry(task, user_id_or_ip)
+                    # reset presented time for projects configured to reset it
+                    project = project_repo.get(project_id)
+                    if not project:
+                        return abort(400)
+                    reset_presented_time = project.info.get("reset_presented_time", False)
+                    if reset_presented_time:
+                        cancelled_task = guard.retrieve_cancelled_timestamp(task, user_id_or_ip)
+                        if cancelled_task:
+                            # user is returning to same task upon cancelling the task
```

Comment:
Does the lock on the task remain even if the user has canceled the task? If not, how can we assume that it is the same user?

Reply:
Its the soft lock that is present on the task upon cancel task. With soft lock, other users can obtain the lock and also if same user returns, the soft lock mechanism presents same task to that user first.